### ISSUE TITLE
[a11y]: Cannot drop elements by ENTER or SPACE key

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/carbon#19365]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.81

### React version

18

### Automated testing tool and ruleset

N/A

### Description

Looks like https://github.com/carbon-design-system/carbon/pull/14320 broke drag-and-drop via keyboard.   You can still start dragging via SPACE or ENTER key, but you can no longer drop via SPACE or ENTER key.

The problem seems to be Modal's call to `event.stopPropagation()` in its `handleKeyDown()` method (introduced in https://github.com/carbon-design-system/carbon/pull/14320).  Because @dnd-kit depends on the SPACE and ENTER key events to bubble to `<body>`.  (I.E. it sets up an keydown listener on `<body>`).

### Reproduction/example

https://carbon-for-ibm-products.netlify.app/?path=/story/deprecated-datagrid-datagrid-columncustomization--column-customization-usage-story&globals=viewport:basic

### Steps to reproduce

1. Navigate to https://carbon-for-ibm-products.netlify.app/?path=/story/deprecated-datagrid-datagrid-columncustomization--column-customization-usage-story&globals=viewport:basic.
2. Click IconButton to open "Customize columns" tearsheet.
3. Tab to "First name" row, and then use SPACE key to start drag-and-drop
4. Optionally use up and/or down arrows to see drag-and-drop working.
5. Try using the SPACE key to end the drag-and-drop.  It doesn't work.

A more complicated problem is if you use the ESC key to try to cancel that drag-and-drop.   That closes the whole modal.  That's harder to fix but it seems like lower priority.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by wkeese** [comment]
This is for @davidmenendez.

**Comment by sangeethababu9223** [comment]
Hey @wkeese ,
I just checked the drag and drop issue in the [Products storybook](https://carbon-for-ibm-products.netlify.app/?path=/story/deprecated-datagrid-columncustomization--column-customization-usage-story)
And it looks like this has been fixed. Please let me know if I've missed something here.

As you mentioned, the issue with pressing `Esc` key closing the modal itself instead of the drag and drop is still there. I could look into this.

**Comment by wkeese** [comment]
Oh you're right, you can end the drop via SPACE key again.

**Comment by sangeethababu9223** [comment]
The drag-and-drop issue related to keyboard interaction appears to have already been resolved.

However, the interaction between Modal and Popover components requires further investigation. A dedicated research issue will be created to thoroughly analyze this combination and identify any potential edge cases or required improvements.

**Comment by sangeethababu9223** [comment]
Closing this issue as the research on Esc key behavior will be continued in: https://github.com/carbon-design-system/carbon/issues/21650
